src/client/Microsoft.Identity.Client/AppConfig/ApplicationConfiguration.cs

```diff
@@ -119,9 +119,18 @@ public string ClientVersion
 
         public bool RetryOnServerErrors { get; set; } = true;
 
+#region ManagedIdentity
         public ManagedIdentityId ManagedIdentityId { get; internal set; }
-
         public bool IsManagedIdentity { get; }
+
+        public CryptoKeyType ManagedIdentityCredentialKeyType { get; internal set; }

```

Comment:
Why do we need to cache these here when you have the `KeyMaterialManager` which gets or creates them?